torchvision/models/vision_transformer.py

```diff
@@ -148,12 +159,32 @@ def __init__(
         self.num_classes = num_classes
         self.representation_size = representation_size
         self.norm_layer = norm_layer
+        self.conv_stem_layers = conv_stem_layers
 
         input_channels = 3
 
-        # The conv_proj is a more efficient version of reshaping, permuting
-        # and projecting the input
-        self.conv_proj = nn.Conv2d(input_channels, hidden_dim, kernel_size=patch_size, stride=patch_size)
+        if self.conv_stem_layers is not None:
+            # As per https://arxiv.org/abs/2106.14881
+            self.conv_proj = nn.Sequential()
+            prev_channels = input_channels
+            for i, conv_stem_layer in enumerate(self.conv_stem_layers):
+                if isinstance(conv_stem_layer, Dict):
+                    conv_stem_layer_params = ConvStemLayer(**conv_stem_layer)._asdict()
+                elif isinstance(conv_stem_layer, ConvStemLayer):
+                    conv_stem_layer_params = conv_stem_layer._asdict()
+                self.conv_proj.add_module(
+                    f"conv_{i}",
+                    nn.Conv2d(
+                        in_channels=prev_channels,
+                        **conv_stem_layer_params,
+                    ),
+                )
+                self.conv_proj.add_module(f"bn_{i}", nn.BatchNorm2d(conv_stem_layer_params["out_channels"]))
+                self.conv_proj.add_module(f"relu_{i}", nn.ReLU())
```

Comment:
We should try to use the common blocks of Vision as much as we can. For Conv-Norm-Activation we have `ConvNormActivation`. See examples on [RegNets](https://github.com/pytorch/vision/blob/e3767f8aa51921460c89e1d396ccce097cebc9d5/torchvision/models/regnet.py#L91).